### ISSUE TITLE
User defined functions and other clarifications

### ISSUE THREAD

**Issue by ashuein** [GiulioRossetti/cdlib#144]
I have several questions which pertain to custom functions for community partition and scoring

1) If I have a graph partition of nodes then is there any way to convert it to a **"CDLIB node clustering"** object so that I can apply CDLIB functions, such as scoring, to it? 

2) Is there any way I can apply multiple scoring functions as evaluation parameters to grid_search or pool ensemble functions in CDLIB? Currently, it seems I can use only one, and to use others I need to run individually for every scoring function. But using multiple algorithms is possible.

3) In the "pool" function under the ensemble approach, how to include community detection algorithms that don't have any parameter to assign such as INFOMAP? Currently, it is required by default that the number of methods and number of configurations must be the same. I think it inclusion of algorithms without any parameter requirement should be allowed with "None" as the definition in the configurations list.

**Comment by GiulioRossetti** [comment]
Hi,
thank you for opening this issue. 

1. Yes, it is quite easy actually (see the [FAQ](https://cdlib.readthedocs.io/en/latest/tutorial.html)). Basically, you only need to shape your clustering as a list of lists of nodes, the build the NodeClustering object out of it, as in this short example:


```
from cdlib import NodeClustering

communities = [[1,2,3], [4,5,6], [7,8,9,10,11]]
coms = NodeClustering(communities, graph=g, method_name="your_method") # g is the graph instance.
```


2. Currently it is not possible to do that: only a single quality score can be used in greed_search/pool. I agree that a multi-criteria search could be a nice thing to have for that methods. I'll see what I can do about it in a next release. 

3. Ok, I have to admit that I poorly described this aspect in the documentation, my fault. You can include methods without parameters in the pool, just specify an empty iterable for them. See this example:

```
g = nx.karate_club_graph()

# Louvain
resolution = ensemble.Parameter(name="resolution", start=0.1, end=1, step=0.1)
randomize = ensemble.BoolParameter(name="randomize")
louvain_conf = [resolution, randomize]

# Infomap
infomap_conf = []

methods = [algorithms.louvain, algorithms.infomap]

for communities in ensemble.pool(g, methods, [louvain_conf, infomap_conf]):
    print(communities.method_name)
```

Hope this helps!

Best,
Giulio

**Comment by ashuein** [comment]
Dear Giulio,

Thanks for the prompt response. The first two points nicely address the issues I raised. However, the third one doesn't work yet.

While trying to pass an empty configuration to the ensemble.pool_grid_filter I get the following error:

`ValueError: parameter should be either an instance of Parameter or of BoolParameter`

What I understand is because it needs the CDLIB parameter instance and not an empty list. So I modified this with

`infomap_conf = ensemble.Parameter()`

But again I get another error:

`TypeError: infomap() keywords must be strings`


What am I missing here?
